cogs/hangman.py

```diff
@@ -0,0 +1,293 @@
+        word_length="The length of the random word as an integer or 'Random' for any word length. (e.g. 7)",
+    )
+    @app_commands.choices(
+        category=[
+            app_commands.Choice(name="Noun", value=1),
+            app_commands.Choice(name="Adjective", value=2),
+            app_commands.Choice(name="Verb", value=3),
+            app_commands.Choice(name="Random", value=4),
+        ],
+    )
+    async def playHangman(
+        self,
+        inter: discord.Interaction,
+        category: app_commands.Choice[int],
+        word_length: str,
+    ):
+        await inter.response.defer()
+        r = RandomWord()
+
+        if category.value == 1:
+            selectedCategory = ["nouns"]
+        elif category.value == 2:
+            selectedCategory = ["adjectives"]
+        elif category.value == 3:
+            selectedCategory = ["verbs"]
+        else:
+            selectedCategory = None
+
+        if word_length.lower() == "random":
+            selectedLength = None
+        else:
+            try:
+                selectedLength = int(word_length)
+            except ValueError:
+                await inter.followup.send(
+                    "I can't generate a word unless word_length is an integer or 'Random', please try again."
+                )
+                return
+
+        try:
+            randomWord = r.word(
+                include_parts_of_speech=selectedCategory,
+                word_min_length=selectedLength,
+                word_max_length=selectedLength,
+            )
+        except Exception as e:
+            await inter.followup.send(
+                "I can't generate a word with that word length, please try again."
+            )
+            return
+
+        embed = discord.Embed(
+            title=f"Hangman!",
+            description=f""
+            "```\n"
+            "   +----+\n"
+            "   |    |\n"
+            "   |   \n"
+            "   |   \n"
+            "   |   \n"
+            "   ===\n"
+            "```",
+            color=discord.Color.orange(),
+        )
+        embed.add_field(name="Category", value=f"{category.name}", inline=False)
+        embed.add_field(name="Incorrect Guesses", value="0/6", inline=True)
+        embed.add_field(name="Incorrect Letters", value=f"None", inline=True)
+        embed.add_field(
+            name="Word",
+            value=f"```{' '.join('_' for c in randomWord)}```",
+            inline=False,
+        )
+
+        view = MyView(id, randomWord, category.name)
+        await inter.followup.send(
+            embed=embed,
+            view=view,
+        )
+
+
+class MyView(discord.ui.View):
+    def __init__(self, creatorId, randomWord, category):
+        super().__init__(timeout=None)
+        self.creatorId = creatorId
+        self.randomWord = randomWord.upper()
+        self.revealedWord = "".join("_" for c in self.randomWord)
+        self.incorrectLetters = []
```

Comment:
snake case

Reply:
Resolving as the project currently uses camelCase.